src/lib/objects/index.d.ts

```diff
@@ -31,22 +31,30 @@ export interface WakuObjectState {
 	readonly profile: User
 	readonly users: User[]
 	readonly tokens: Token[]
+	readonly chatName: string
 }
 
-type StoreType = JSONSerializable
-type DataMessageType = JSONSerializable
-
-export interface WakuObjectContext extends WakuObjectAdapter {
+export interface WakuObjectContext<
+	StoreType extends JSONSerializable = JSONSerializable,
+	DataMessageType extends JSONSerializable = JSONSerializable,
+	ViewType extends string = string,
+> extends WakuObjectAdapter {
 	readonly store?: StoreType
 	updateStore: (updater: (state?: StoreType) => StoreType) => void
 
 	send: (data: DataMessageType) => Promise<void>
 
-	readonly view?: string
-	onViewChange: (view: string) => void
+	readonly view?: ViewType // Screen to show on Waku object
+	readonly params: string[] // Other path params after the screen
```

Comment:
If I understand correctly this is used so that the individual screens can be addressed directly with URLs.

If that's the case I would rather call this `viewParams` because otherwise the naming is not very clear, or at least for me it is puzzling when I see `args.params` in the code.